.github/workflows/windows.yml

```diff
@@ -27,7 +27,9 @@ jobs:
         - { name: Release }
         build-system:
         - "Visual Studio 17 2022"
-        - "Ninja"
+        # Ninja is disabled because the standard library implementation is uses does not
```

Comment:
Build system does not implie the use of a specific standard library.
When you use Ninja as generator you have to define the compilers and toolset in case you want MSVC:
cmake -G "Ninja" -T "v142" -DCMAKE_C_COMPILER="C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe" -DCMAKE_CXX_COMPILER="C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe"

Reply:
I should have worded this comment differently, indeed.

I am not knowledgeable of Windows' toolchains at all. Feel free to adapt it and the toolchain as you wish.

Reply:
See the `Setup clang` step for how clang is used with both ninja and msbuild.

Reply:
> I am not knowledgeable of Windows' toolchains at all.

To be short: 
- Windows as an ABI which is usable with libstd-c++, libc++, and msvc-stl , which are respectively provided by the *build toolchains* of mingw, llvm/clang and msvc (cl is the compiler, msvc is the toolchain). 
- each *build toolchain* has a prefered stl/runtime implementation by default, it's changed only if specified explicitly: 
    -gcc/mingw will use libstd-c++ by default
    - clang will use msvc-stl by default (even if libc++ is installed with it, it has to be requested explicitly to be used)
    - msvc can only use msvc-stl
- the *build system* is irrelevant to the choice of the build toolchain, however they have different defaults if you dont specificy the build-toolchain:
    - msbuild (generator "Visual Studio blah") will use msvc by default, therefore msvc-stl
    - ninja will use mingw by default, therefore libstdc++
- finally, cmake also has a default generator which depends on the development platform:
    - on windows, if not specified, it will use the "Visual Studio ..." generator with everything by default
    - on linux it will use the "Make" generator by default
    - not sure on MacOSX but I believe it generates something known by XCode by default
    - AFAIK Ninja is never a default generator
    - using either ninja or make or similar agnostic build-systems implies probably to also specify which compiler (and therfore build toolchain and therefore stl and runtime impl) to use